backend/src/main/java/com/soen341/instagram/service/impl/PictureService.java

```diff
@@ -133,4 +150,54 @@ private BufferedImage compressImage(BufferedImage image) {
         compressedImage.createGraphics().drawImage(image, 0, 0, Color.WHITE, null);
         return compressedImage;
     }
+    
+    // like service
+	public int likePicture(final long pictureId) {
+		final Picture picture = getPictureFromId(pictureId);
+		final Set<Account> likedBy = picture.getLikedBy();
+		final boolean liked = likedBy.add(getCurrentUser());
+		
+		if (!liked) {
+			throw new MultipleLikeException("A picture can only be liked once by the same user.");
+		}
+		
+		pictureRepository.save(picture);
+		
+		return picture.getLikeCount();
+		
+	}
+	
+	public int unlikePicture(final long pictureId) {
+		final Picture picture = getPictureFromId(pictureId);
+		final Set<Account> likedBy = picture.getLikedBy();
+		final boolean unliked = likedBy.remove(getCurrentUser());
+		
+		if (!unliked) {
+			throw new NoLikeException("This picture has not yet been liked by the user.");
+		}
+		
+		pictureRepository.save(picture);
+		
+		return picture.getLikeCount();
+		
+	}
+	
+	private Account getCurrentUser()
```

Comment:
This method is inefficient (makes a query for nothing).
Anyway, you should remove this method because it already exists in `backend/src/main/java/com/soen341/instagram/utils/UserAccessor.java`. You should instead inject it with `@Autowired` it and use its method

Reply:
I do not have utils in my branch, when was that implemented?